### PR TITLE
NIAD-1096: Add participant element when translating Immunization to ObservationStatement

### PR DESCRIPTION
**Pull request by andrewwa-kainos**
https://gpitbjss.atlassian.net/browse/NIAD-1096

AC3 - add Participant element to ObservationStatement

GIVEN Immunization is translated to ObservationStatement
THEN Practitioner element is added to ObservationStatement
AND Practitioner is referencing Agent

### CODE REVIEW

File: service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ImmunizationObservationStatementMapperTest.java [hunk 1]
```diff
@@ -95,7 +97,9 @@ public void tearDown() {
     @ParameterizedTest
     @MethodSource("resourceFileParams")
     public void When_MappingImmunizationJson_Expect_ObservationStatementXmlOutput(String inputJson, String outputXml,
-        boolean isNested) throws IOException {
+                                                                                  boolean isNested) throws IOException {
+        messageContext.getIdMapper().getOrNew(ResourceType.Practitioner, "6D340A1B-BC15-4D4E-93CF-BBCB5B74DF73");
```
Comment (n-r-b) [inline service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ImmunizationObservationStatementMapperTest.java:101]: I don't think we have any tests covering the scenarios:
- immunization without a Practitioner
- immunization with a Practitioner, but without an actor

**Review by msolecka — DISMISSED** [review]

File: service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ImmunizationObservationStatementMapperTest.java [hunk 2]
```diff
@@ -51,6 +52,10 @@
         + "immunization-reason-not-given-text.json";
     private static final String INPUT_JSON_WITH_VACCINE_CODE = IMMUNIZATION_FILE_LOCATIONS
         + "immunization-vaccine-code-given.json";
+    private static final String INPUT_JSON_WITHOUT_PRACTITIONER = IMMUNIZATION_FILE_LOCATIONS
+        + "immunization-no-practitioner.json";
+    private static final String INPUT_JSON_WITHOUT_PRACTITIONER_ACTOR = IMMUNIZATION_FILE_LOCATIONS
+        + "immunization-practitioner-but-no-actor.json";
```
Comment (matthewdoreilly) [inline service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ImmunizationObservationStatementMapperTest.java:58]: What about the scenario if an `immunization` has a `practitioner` with an `actor`?

File: service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ImmunizationObservationStatementMapperTest.java [hunk 3]
```diff
@@ -51,6 +52,10 @@
         + "immunization-reason-not-given-text.json";
     private static final String INPUT_JSON_WITH_VACCINE_CODE = IMMUNIZATION_FILE_LOCATIONS
         + "immunization-vaccine-code-given.json";
+    private static final String INPUT_JSON_WITHOUT_PRACTITIONER = IMMUNIZATION_FILE_LOCATIONS
+        + "immunization-no-practitioner.json";
+    private static final String INPUT_JSON_WITHOUT_PRACTITIONER_ACTOR = IMMUNIZATION_FILE_LOCATIONS
```
Comment (matthewdoreilly) [inline service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ImmunizationObservationStatementMapperTest.java:57]: Think this should be renamed `INPUT_JSON_WITH_PRACTITIONER_BUT_NO_ACTOR`

File: service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ImmunizationObservationStatementMapperTest.java (hunk 2 above)
Comment (andrewwa-kainos) [reply service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ImmunizationObservationStatementMapperTest.java:58]: That's covered by existing tests - e.g. immunization-all-pertinent-information.json

File: service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ImmunizationObservationStatementMapperTest.java (hunk 3 above)
Comment (andrewwa-kainos) [reply service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ImmunizationObservationStatementMapperTest.java:57]: Sure

**Review by mattd-kainos — APPROVED** [review]

**Review by msolecka — DISMISSED** [review]

**Review by n-r-b — APPROVED** [review]

**Review by Broom94 — APPROVED** [review]

**Review by msolecka — APPROVED** [review]

**Pull request merged**